core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/cols.kt

```diff
@@ -0,0 +1,967 @@
+     *
+     * `df.`[select][DataFrame.select]` { `[all][all]`()`[`[`][ColumnSet.cols]`1`[..][Int.rangeTo]`5`[`]`][ColumnSet.cols]` }`
+     */
+    private interface ColumnSetColsRangeDocs
+
+    /** @include [ColumnSetColsRangeDocs] */
+    @Suppress("UNCHECKED_CAST")
+    public fun <C> ColumnSet<C>.cols(range: IntRange): ColumnSet<C> = colsInternal(range) as ColumnSet<C>
+
+    /** @include [ColumnSetColsRangeDocs] */
+    public operator fun <C> ColumnSet<C>.get(range: IntRange): ColumnSet<C> = cols(range)
+
+    /**
+     * @include [CommonColsRangeDocs]
+     * @arg [CommonColsRangeDocs.ExampleArg]
+     *
+     * `df.`[select][DataFrame.select]` { `[cols][ColumnsSelectionDsl.cols]`(1`[..][Int.rangeTo]`3) }`
+     */
+    private interface ColumnsSelectionDslColsRangeDocs
+
+    /** @include [ColumnsSelectionDslColsRangeDocs] */
+    public fun ColumnsSelectionDsl<*>.cols(range: IntRange): ColumnSet<*> = this.asSingleColumn().colsInternal(range)
+
+    /**
+     * @include [CommonColsRangeDocs]
+     * @arg [CommonColsRangeDocs.ExampleArg]
+     *
+     * `df.`[select][DataFrame.select]` { myColumnGroup.`[cols][SingleColumn.cols]`(1`[..][Int.rangeTo]`3) }`
+     */
+    private interface SingleColumnColsRangeDocs
+
+    /** @include [SingleColumnColsRangeDocs] */
+    public fun SingleColumn<DataRow<*>>.cols(range: IntRange): ColumnSet<*> = ensureIsColGroup().colsInternal(range)
+
+    /**
+     * @include [CommonColsRangeDocs]
+     * @arg [CommonColsRangeDocs.ExampleArg]
+     *
+     * `df.`[select][DataFrame.select]` { "myColGroup".`[cols][String.cols]`(1`[`..`][Int.rangeTo]`3) }`
+     */
+    private interface StringColsRangeDocs
+
+    /** @include [StringColsRangeDocs] */
+    public fun String.cols(range: IntRange): ColumnSet<*> = columnGroup(this).cols(range)
+
+    /**
+     * @include [CommonColsRangeDocs]
+     * @arg [CommonColsRangeDocs.ExampleArg]
+     *
+     * `df.`[select][DataFrame.select]` { Type::myColumnGroup.`[cols][SingleColumn.cols]`(1`[`..`][Int.rangeTo]`3) }`
+     *
+     */
+    private interface KPropertyColsRangeDocs
+
+    /** @include [KPropertyColsRangeDocs] */
+    public fun KProperty<*>.cols(range: IntRange): ColumnSet<*> = columnGroup(this).cols(range)
+
+    /**
+     * @include [CommonColsRangeDocs]
+     * @arg [CommonColsRangeDocs.ExampleArg]
+     *
+     * `df.`[select][DataFrame.select]` { "pathTo"["myColGroup"].`[cols][ColumnPath.cols]`(0`[..][Int.rangeTo]`1) }`
+     */
+    private interface ColumnPathColsRangeDocs
+
+    /** @include [ColumnPathColsRangeDocs] */
+    public fun ColumnPath.cols(range: IntRange): ColumnSet<*> = columnGroup(this).cols(range)
+
+    /**
+     * ## Columns by Index Range from List of Columns
+     * Helper function to create a [ColumnSet] from a list of columns by specifying a range of indices.
+     *
+     * {@comment TODO find out usages of this function and add examples}
```

Comment:
@koperagen any idea what this was for? It's present but never used. The only place I see the `List<DataColumn<C>>` return type is inside the `DataFrame` interface... any ideas?